src/runner/browser-env/vite/plugins/resolve-module-paths.ts

```diff
@@ -0,0 +1,36 @@
+import url from "node:url";
+import path from "node:path";
+
+import { getImportMetaUrl } from "../utils";
+import { MODULE_NAMES } from "../constants";
+
+import type { Plugin } from "vite";
+import type { VitePluginOptions } from "../types";
+
+export const plugin = (options: VitePluginOptions): Plugin[] => {
+    const dirname = url.fileURLToPath(new URL(".", getImportMetaUrl(__filename)));
+
+    const browserModulesPath = path.resolve(dirname, "..", "browser-modules");
+    const browserRunnerModulePath = path.resolve(browserModulesPath, "index.js");
+    const globalsModulePath = path.resolve(browserModulesPath, "globals.js");
+
+    return [
+        {
+            name: "hermione:resolveModulePaths",
+            enforce: "pre",
+            resolveId: (id): string | void => {
+                if (id.endsWith(MODULE_NAMES.browserRunner)) {
+                    return browserRunnerModulePath;
+                }
+
+                if (id.endsWith(MODULE_NAMES.globals)) {
+                    return globalsModulePath;
+                }
+
+                if (id.endsWith(MODULE_NAMES.mocha)) {
+                    return options.modulePaths.mocha;
```

Comment:
hm, isn't it a bit weird that some resolutions are hard-coded here, but some are passed as options, but their types are still hard-coded? Maybe we should stick to the single approach (include all resolutions here or move them all to options)?

Also, thinking out loud, do you think it's worth it to have a separate plugin for such resolutions? An alternative would be to just write complete paths at the generate-index-html helper, which sounds easier? But I may be missing something.

Not critical for me. Leaving this up to you.

Reply:
I agree it wasn't the best decision. Fixed.